announcements/announcements.go

```diff
@@ -115,11 +112,52 @@ func remove(im *imap.Dialer, uids []int) error {
 	return err
 }
 
-func formatContents(text string) string {
-	cutoff := strings.LastIndex(text, "\\-\\-")
-	if cutoff == -1 {
-		return text
+var replacer = strings.NewReplacer("\n\n", "\n")
+
+// formatAnnouncement formats an announcement from the given subject and contents
+//
+// it does a slightly ugly hack with maxLen to both make the announcement fit and pretty, so try to keep some leeway.
+func formatAnnouncement(maxLen int, subject string, body string) Announcement {
+	body = replacer.Replace(body)
+	body = strings.TrimSpace(body)
+
+	cutoff := strings.LastIndex(body, "\\-\\-")
+	if cutoff != -1 {
+		body = body[:cutoff]
+		body = strings.TrimSpace(body)
+	}
+
+	var contents []string
+
+	currLen := len(subject) + len(body)
```

Comment:
I think I kind of understand what you're trying to do here, but it frankly comes off as too over engineered for what actually needs to be done;

While the message exceeds `maxLen` characters, just get the last index of `\n` and trim there. If there's no `\n`, just trim to `maxLen` and carry on. I think that's honestly more than good enough.

Reply:
If that seems too complicated, throwing away the rest of the stuff is fine too

Reply:
thrown away